compute_endpoint/globus_compute_endpoint/MULTIUSER_SETUP.md

```diff
@@ -0,0 +1,334 @@
+id), then a user may specify that via the `user_endpoint_config` argument to
+the Executor constructor:
+
+```python
+from globus_compute_sdk import Executor
+
+def some_task(*a, **k):
+    return 1
+
+mep_site_id = "..."  # as acquired from step 1
+with Executor(
+    endpoint_id=mep_site_id,
+    user_endpoint_config={"account_id": "user_allocation_account_id"},
+) as ex:
+    fut = ex.submit(some_task)
+    print("Result:", fut.result())  # Reminder: blocks until result received
+```
+
+## Key Benefits
+
+### For Administrators
+
+This biggest benefit of a MEP setup is a lowering of the barrier for legitimate
+users of a site.  To date, knowledge of the command line has been critical to
+most users of High Performance Computing (HPC)) systems, though only as a
+necessity of infrastructure rather than a legitimate scientific purpose.  A MEP
+allows a user to ignore many of the important-but-not-really details of
+plumbing, like logging in through SSH, restarting user-only daemons, or, in the
+case of Globus Compute, fine-tuning a scheduler options by managing multiple
+endpoint configurations.  The only thing they need to do is run their scripts
+locally on their own workstation, and the rest "just works."
+
+Another boon for administrators is the ability to fine-tune and pre-configure
+what resources UEPs may utilize.  For example, many users struggle to discover
+which interface is routed to a cluster's internal network; the administrator
+can preset that, completely bypassing the question.  Using ALCF's Polaris as an
+example, the administrator could use the following user configuration template
+(`user_config_template.yaml`) to place all jobs sent to this MEP to the
+`debug-scaling` queue, and pre-select the obvious defaults ([per the
+documentation](https://docs.alcf.anl.gov/polaris/running-jobs/)):
+
+```yaml
+display_name: Polaris at ALCF - debug-scaling queue
+engine:
+  type: GlobusComputeEngine
+  address:
+    type: address_by_interface
+    ifname: bond0
+
+  strategy:
+    type: SimpleStrategy
+    max_idletime: 30
+
+  provider:
+    type: PBSProProvider
+    queue: debug-scaling
+
+    account: {{ ACCOUNT_ID }}
+
+    # Command to be run before starting a worker
+    # e.g., "module load Anaconda; source activate parsl_env"
+    worker_init: {{ WORKER_INIT_COMMAND|default() }}
+
+    init_blocks: 0
+    min_blocks: 0
+    max_blocks: 1
+    nodes_per_block: {{ NODES_PER_BLOCK|default(1) }}
+
+    walltime: 1:00:00
+
+    launcher:
+      type: MpiExecLauncher
+```
+
+The user must specify the `ACCOUNT_ID`, and could optionally specify the
+`WORKER_INIT_COMMAND` and `NODES_PER_BLOCK` variables.  If the user's jobs
+finish and no more work comes in after `max_idletime` seconds (30s), the UEP
+will scale down and consume no more wall time.
+
+(Validation of the user-provided variables is available, but not yet
+documented.  For the motivated, please look to `user_config_schema.json`.)
```

Comment:
This seems like a good place to elaborate a bit more on how to modify the user config schema to perform validation.

Reply:
I'd like to push that off for the fuller documentation.  The ticket in question is about a script for early adopters to setup a MEP on an EC2 instance.  This basic set of instructions is yet a singular file in the repo, and not linked to anything, other than as an ad-hoc set of instructions for early adopters.  When we incorporate it into the wider documentation, let's totally do this.